File: Airfare Grants/airfaregrants.cpp
#include <iostream>
using namespace std;
int main()
{
    int n;
    int smallest = 1000000;
    int largest = 0;
    int cost = 0;
    cin >> n;
    cin.ignore();

    for (int i = 0; i < n; i++)
    {
        int value;
        cin >> value;
        if (value < smallest)
        {
            smallest = value;
        }
        if (value > largest)
        {
            largest = value;
        }
    }

    int reimbursed = largest / 2;
    cost = smallest - reimbursed;

    if (cost < 0)
    {
        cout << 0;
    }
    else
    {
        cout << cost;
    }

    return 0;
}

File: Airfare Grants/airfaregrants.py
n = int(input())
smallest = float('inf')
largest, cost = 0,0
for i in range(n):
    value = int(input())
    smallest = min(smallest, value)
    largest = max(largest, value)

reimbursed = largest // 2
cost = smallest - reimbursed

if cost < 0:
    print(0)
else:
    print(cost)